graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -26,6 +42,58 @@ def _is_fractional(selectivity):
     return selectivity.kind == FRACTIONAL_SELECTIVITY
 
 
+def _get_intersection_of_IntegerIntervals(interval_a, interval_b):
```

Comment:
There's a lot of complexity and room for error in the implementation. Is it simpler to define `stronger_lower_bound` and `sronger_upper_bound` functions, and return `(stronger_lower_bound(a.lower_bound, b.lower_bound), stronger_upper_bound(a.upper_bound, b.upper_bound)`, making sure to return `None` if the resulting interval is empty? Feel free to argue if it doesn't look simpler to you.

Reply:
Thanks for the suggestion!